main.rb

```diff
@@ -0,0 +1,71 @@
+require 'open3'
+
+def get_env_variable(key)
+  return (ENV[key] == nil || ENV[key] == "") ? nil : ENV[key]
+end
+ 
+def env_has_key(key)
+	value = get_env_variable(key)
+	return value unless value.nil? || value.empty?
+
+	abort("Input #{key} is missing.")
+end
+
+def run_command(command)
+	puts "@@[command] Executing: #{command}"
+	stdout_str, stderr_str, status = Open3.capture3(command)
+
+	if status.success?
+		puts stdout_str unless stdout_str.empty?
+		puts stderr_str unless stderr_str.empty?
+		return stderr_str.empty? ? stdout_str : stderr_str
+	else
+		puts "@@[error] Command failed:\n#{stderr_str}"
+		raise stderr_str
+	end
+end
+
+def abort_with0(message)
+  puts "@@[error] #{message}"
```

Comment:
Only for suggestion!

If current version and selected version are equal, you are finishing program with exit0. But you think, is this an error? or is this a warning? I think this is a warning. Two versions are equal, I do nothing. 

When you use this one, the step will shown success in the pipeline step list, but in details of step it will be shown red error message. 

As I sad, only suggestion.

Reply:
We can consult to @csonuryilmaz , but it is not a big deal.